### ISSUE TITLE
[BUG] Fixing `no-unnecessary-arbitrary-value` causes a `no-custom-classname`

### ISSUE THREAD

**Issue by sjoqvist** [francoismassart/eslint-plugin-tailwindcss#469]
**Describe the bug**
The plugin reports **No need for arbitrary 'leading-[1.125]', use 'leading-1.125' instead** (`tailwindcss/no-unnecessary-arbitrary-value`), but fixing it causes **Classname 'leading-1.125' is not a Tailwind CSS class** (`tailwindcss/no-custom-classname`). Furthermore, styling changes as Tailwind CSS doesn't appear to generate code after the syntax fix.

**To Reproduce**

1. In a new folder, set up the following files.
   ```js
   // eslint.config.js
   const react = require('eslint-plugin-react');
   const tailwindcss = require('eslint-plugin-tailwindcss');

   module.exports = [
     react.configs.flat['recommended'],
     tailwindcss.configs.recommended,
   ];
   ```
   ```jsx
   // src/Component.jsx
   const React = require('react');

   const Component = () => (<div className="leading-[1.125]">Hello World!</div>);
   ```
   ```css
   /* src/style.css */
   @import 'tailwindcss';
   ```
2. Run
   ```sh
   npm init -y
   npm i -DE eslint@9.39.5 eslint-plugin-react@7.37.5 eslint-plugin-tailwindcss@4.2.0
   ```
3. Run
   ```sh
   npx eslint src/Component.jsx
   ```
   and pay attention to the error
   ```
   /tmp/bug/src/Component.jsx
     4:42  warning  No need for arbitrary 'leading-[1.125]', use 'leading-1.125' instead  tailwindcss/no-unnecessary-arbitrary-value
   ```
4. Run
   ```sh
   npx eslint src/Component.jsx --fix
   ```
   and note that there's now a new error:
   ```
   /tmp/bug/src/Component.jsx
     4:42  warning  Classname 'leading-1.125' is not a Tailwind CSS class  tailwindcss/no-custom-classname
   ```

**Expected behavior**
The rule should either not complain about `leading-[1.125]`, or change it into something that's a valid Tailwind CSS class.

**Environment (please complete the following information):**

- OS: Windows 11 Pro with Ubuntu 24.04 in WSL
- Softwares + version used:
  - GNU bash 5.2.21(1)-release
  - node 26.5.0
  - npm 12.0.1
  - eslint 9.39.5
  - eslint-plugin-react 7.37.5
  - eslint-plugin-tailwindcss 4.2.0

**Comment by mikemikimike** [comment]
Implemented in [PR #482](https://github.com/francoismassart/eslint-plugin-tailwindcss/pull/482).

The rule now keeps decimal arbitrary values when their normalized class name is not valid in Tailwind CSS v4, while preserving existing integer and theme preset suggestions. A regression test covers `leading-[1.125]` so it is no longer changed into the invalid `leading-1.125` class.

Local verification completed with the targeted rule tests, repository lint, the package build, and a built-package ESLint smoke test.
